### ISSUE TITLE
webpush support ?

### ISSUE THREAD

**Issue by sabrinatseng** [appleboy/gorush#248]

**Comment by appleboy** [comment]
See the PR. #310

**Comment by appleboy** [comment]
I will implement this feature using https://github.com/SherClockHolmes/webpush-go

**Comment by pwFoo** [comment]
PR #310 is still open? So not yet implemented?

Searching for a push notification service / server to separate push service from the application (backend / server part).

**Comment by slimus** [comment]
PR #310  is outdated. We should close it and create the new one with actual weboush lib

**Comment by pwFoo** [comment]
Webpush lib would be awesome!!!

Generate vapid keys and take care about communication to the push services of Microsoft, Mozilla, Apple and Google dependent on used client device / browser. :+1:

**Comment by joe-getcouragenow** [comment]
I am using Flutter and can test this with it as flutter web has this ability these days

**Comment by appleboy** [comment]
I will close the PR #310 since of outdated.

**Comment by truong-hua** [comment]
This would be a great feature

**Comment by gedw99** [comment]
@appleboy https://github.com/binwiederhier/ntfy is a web push backend and Web Client that works on all web clients and is installable as a PWA with A2HS working well.

https://github.com/binwiederhier/ntfy/blob/main/go.mod

It looks like an excellent basis.

The web G>UI is through too.

**Comment by appleboy** [comment]
@gedw99 Thanks for your information. I will check the https://github.com/SherClockHolmes/webpush-go repo.

**Comment by josejuanmontiel** [comment]
Hi! Seem that from July 24 google deprecate the old api ... and now (i'm new with webpush now) seems not work like before with google chrome (still work with firefox... no checked with other) ... i have a work around for webpush-go library ... commented here https://github.com/SherClockHolmes/webpush-go/issues/82 ... could be interesting retake this

**Comment by appleboy** [comment]
@josejuanmontiel Any Google announcement link?

**Comment by josejuanmontiel** [comment]
At the beginning of this page https://firebase.google.com/docs/cloud-messaging/migrate-v1 ???

**Comment by appleboy** [comment]
Is there no news about removing the web push feature in the link?

**Comment by josejuanmontiel** [comment]
No, but change the way of using... I only probe by now the library suggested in this thread (and there commented the changes to use it for subscription generated in chrome) ... What I'm saying here is adding information if the old PR is retaking with chrome won't work (or this is what happened to me)

**Comment by josejuanmontiel** [comment]
More info about the actual way of use here https://developer.chrome.com/blog/web-push-interop-wins#introducing_vapid_for_server_identification
